### PR TITLE
Add `slideend` event and set/get slider position

### PR DESCRIPTION
**Pull request by cmtoomey**
This PR adds the following

- Public currentPosition tracking (this is a clone of the internal _x value)
- Public setSlider function to programatically set slider position
- Public `slideend` event listener to allow users to track when the slider is being moved, and surface the `currentPosition` if necessary.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ function Compare(a, b, options) {
   }
 
   this._swiper.addEventListener('mousedown', this._onDown);
-  this._swiper.addEventListener('touchstart', this._onDown);
+  this._swiper.addEventListener('touchstart', this._onDown);
```
Comment (tristen) [inline index.js]: nit: extra whitespace on this line

File: index.js [hunk 2]
```diff
@@ -61,6 +61,7 @@ Compare.prototype = {
     this._container.style.WebkitTransform = pos;
     this._clippedMap.getContainer().style.clip = 'rect(0, 999em, ' + this._bounds.height + 'px,' + x + 'px)';
     this._x = x;
+    this.currentPosition = this._x;
```
Comment (tristen) [inline index.js]: Can `this.currentPosition` just replace `this._x`? I don't two named variables of this value is necessary?

File: index.js [hunk 3]
```diff
@@ -87,6 +88,18 @@ Compare.prototype = {
     if (x < 0) x = 0;
     if (x > this._bounds.width) x = this._bounds.width;
     return x;
+  },
+
+  setSlider: function(x) {
+    this._setPosition(x);
+  },
+
+  on: function(event, func) {
```
Comment (tristen) [inline index.js]: I think you'll want to use a different pattern to emit events. [Here's a fairly simple example](https://github.com/tristen/pencil/blob/331dd2fbba6b165670f3085a931de5786cff2ca3/index.js) that uses `EventEmitter` with `on`, `off`, & `fire` methods to control event behaviour. So when a [line like this](https://github.com/mapbox/mapbox-gl-compare/blob/4e54668478436bd9fccaa48cfaf737e833ef39e1/index.js#L75-L78) is called, you would call something like

```js
this.fire('slideend', { currentPosition: this.currentPosition });
```

Comment (cmtoomey) [reply index.js]: Updated in 76704ad

**Comment by tristen** [comment]
@cmtoomey this is looking good! can you add a test to https://github.com/mapbox/mapbox-gl-compare/blob/master/test/index.js that asserts calling `setSlider` updates `currentPosition`?

File: README.md [hunk 4]
```diff
@@ -39,7 +57,7 @@ You'll need a [Mapbox access token](https://www.mapbox.com/help/create-api-acces
 
 Tests require an MapboxAccessToken env variable to be set.
 
-    export MapboxAccessToken="YOUR ACCESS TOKEN"
+    export MapboxAccessToken='YOUR ACCESS TOKEN'
```
Comment (tristen) [inline README.md]: I think we want to keep this double-quoted

File: README.md [hunk 5]
```diff
@@ -1,5 +1,4 @@
-mapbox-gl-compare
----
+## mapbox-gl-compare
```
Comment (tristen) [inline README.md]: Why this change? 

```md
mapbox-gl-compare
---
```

should effectively be the same thing

File: test/index.js [hunk 6]
```diff
@@ -18,7 +18,7 @@ test('Compare', function(t) {
     style: 'mapbox://styles/mapbox/dark-v8'
   });
 
-  new mapboxgl.Compare(a, b);
+  var compare = new mapboxgl.Compare(a, b);
```
Comment (tristen) [inline test/index.js]: nit: use `const` here!

File: README.md (hunk 5 above)
Comment (cmtoomey) [reply README.md]: No idea - and I think Prettier made that change - it's reverted in next commit
